### DISCUSSION TITLE
Introduce a script/tool for checking Renovate prerequisite versions/memory size

### DISCUSSION THREAD

**Discussion by olegkrivtsov** [renovatebot/renovate#31824] [Suggest an Idea]
### What would you like Renovate to be able to do?

While installing Renovate on my Ubuntu 20.04 virtual machine and Ubuntu 18.04 VM, I faced with some issues:

- GIT version was outdated on Ubuntu 18.04
- memory size was insufficient of Ubuntu 20.04 (1GB)

For me as a newbie, it wasn't obvious how to fix those issues, so I had to submit a bug report. 

Suggestion: 

Introduce a bash/python script which would check the following (but not limited):
 - GIT version is compatible
 - C++ compiler is compatible
 - TypeScript version is compatible
 - memory (RAM) size is sufficient
 - maybe something else, e.g. disk space is enough

Add a paragraph to the Renovate documentation suggesting running this tool before/after installation to check prerequisites. I've used a script like that with a PHP framework called Yii2, and it was really useful. 

### If you have any ideas on how this should be implemented, please tell us here.

I guess a Python or Bash script would be sufficient. It could be placed to the root directory of the Renovate installation. The user would just run the script to check prerequisites were installed correctly.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by olegkrivtsov** [comment]
Additional note:

Please maybe mention in your documentation which Git versions are acceptable. Right now it just says "Git" without mentioning the compatible versions of it: https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md#prerequisites

**Comment by viceice** [comment]
* typescript versions is installed by yarn, so no need to validate
* C++ compiler isn't needed
* Script should work for all platforms (linux / windows / macos)
* Script should be in `tools` directory
* I think it should be a simple esm `js` file, check tools directory for samples

**Comment by viceice** [comment]
related to #11167

**Comment by HonkingGoose** [comment]
I think `@viceice` already said what the "recipe" is for the changes. So I'd say we're `status:ready` and can maybe assign `@olegkrivtsov` to this issue, as they volunteered to make the feature for us?

**Comment by rarkins** [comment]
Could this may be be `renovate --diagnose`? So that it doesn't need to be a new installed binary and also can be used even when renovate is installed via npm.
